core/web/p2p_keys_controller_test.go

```diff
@@ -0,0 +1,132 @@
+
+	response, cleanup := client.Get("/v2/p2p_keys")
+	defer cleanup()
+	cltest.AssertServerResponse(t, response, http.StatusOK)
+
+	err := web.ParseJSONAPIResponse(cltest.ParseResponseBody(t, response), &p2pKeys)
+	assert.NoError(t, err)
+
+	require.Len(t, p2pKeys, len(keys))
+
+	assert.Equal(t, keys[0].ID, p2pKeys[0].ID)
+	assert.Equal(t, keys[0].PubKey, p2pKeys[0].PubKey)
+	assert.Equal(t, keys[0].PeerID, p2pKeys[0].PeerID)
+}
+
+func TestP2PKeysController_Create_InvalidBody(t *testing.T) {
+	client, _, cleanup := setupP2PKeysControllerTests(t)
+	defer cleanup()
+
+	invalidBody, _ := json.Marshal(struct {
+		BadParam string
+	}{
+		BadParam: "randomString",
+	})
+	response, cleanup := client.Post("/v2/p2p_keys", bytes.NewBuffer(invalidBody))
+	defer cleanup()
+	cltest.AssertServerResponse(t, response, http.StatusUnprocessableEntity)
+}
+
+func TestP2PKeysController_Create_HappyPath(t *testing.T) {
+	client, OCRKeyStore, cleanup := setupP2PKeysControllerTests(t)
+	defer cleanup()
+
+	keys, _ := OCRKeyStore.FindEncryptedP2PKeys()
+	initialLength := len(keys)
+
+	body, _ := json.Marshal(models.CreateP2PKeysRequest{
+		Password: cltest.Password,
+	})
+	response, cleanup := client.Post("/v2/p2p_keys", bytes.NewBuffer(body))
+	defer cleanup()
+	cltest.AssertServerResponse(t, response, http.StatusOK)
+
+	keys, _ = OCRKeyStore.FindEncryptedP2PKeys()
+	require.Len(t, keys, initialLength+1)
+
+	encryptedP2PKey := p2pkey.EncryptedP2PKey{}
+	err := web.ParseJSONAPIResponse(cltest.ParseResponseBody(t, response), &encryptedP2PKey)
+	assert.NoError(t, err)
+
+	lastKeyIndex := len(keys) - 1
+	assert.Equal(t, keys[lastKeyIndex].ID, encryptedP2PKey.ID)
+	assert.Equal(t, keys[lastKeyIndex].PubKey, encryptedP2PKey.PubKey)
+	assert.Equal(t, keys[lastKeyIndex].PeerID, encryptedP2PKey.PeerID)
+
+	_, exists := OCRKeyStore.DecryptedP2PKey(peer.ID(encryptedP2PKey.PeerID))
+	assert.Equal(t, exists, true)
+}
+
+func TestP2PKeysController_Delete_InvalidP2PKey(t *testing.T) {
+	client, _, cleanup := setupP2PKeysControllerTests(t)
+	defer cleanup()
+
+	invalidP2PKeyID := "bad_key_id"
+	response, cleanup := client.Delete("/v2/p2p_keys/" + invalidP2PKeyID)
+	defer cleanup()
+	assert.Equal(t, http.StatusUnprocessableEntity, response.StatusCode)
+}
+
+func TestP2PKeysController_Delete_NonExistentP2PKeyID(t *testing.T) {
+	client, _, cleanup := setupP2PKeysControllerTests(t)
+	defer cleanup()
+
+	nonExistentP2PKeyID := "1234567890"
+	response, cleanup := client.Delete("/v2/p2p_keys/" + nonExistentP2PKeyID)
+	defer cleanup()
+	assert.Equal(t, http.StatusNotFound, response.StatusCode)
+}
+
+func TestP2PKeysController_Delete_HappyPath(t *testing.T) {
+	client, OCRKeyStore, cleanup := setupP2PKeysControllerTests(t)
+	defer cleanup()
+
+	keys, _ := OCRKeyStore.FindEncryptedP2PKeys()
+	initialLength := len(keys)
+	_, encryptedKeyBundle, _ := OCRKeyStore.GenerateEncryptedP2PKey(cltest.Password)
+
+	response, cleanup := client.Delete("/v2/p2p_keys/" + encryptedKeyBundle.GetID())
```

Comment:
You can also use `fmt.Sprintf` here too

Reply:
I'll leave this as is given that `fmt` is not a dep in this test and `.GetID()` calls it under the hood anyways.